backend/src/permissions/permissions.service.spec.ts

```diff
@@ -712,6 +714,117 @@ describe('PermissionsService', () => {
+        mediaUpload.note = Promise.resolve(notes[1]);
+        mediaUpload.user = Promise.resolve(user1);
+        expect(
+          service.checkMediaDeletePermission(user2, mediaUpload),
+        ).toBeTruthy();
+      });
+    });
+    describe('denies', () => {
+      it('for not owner', async () => {
+        const user3 = {} as User;
+        user3.id = 3;
+        const mediaUpload = {} as MediaUpload;
+        mediaUpload.note = Promise.resolve(notes[1]);
+        mediaUpload.user = Promise.resolve(user1);
+        console.log(user3, mediaUpload.user, (await mediaUpload.note)?.owner);
+        expect(
+          await service.checkMediaDeletePermission(user3, mediaUpload),
+        ).toBeFalsy();
+      });
+    });
+  });
+
+  describe('checkPermissionOnNote', () => {
+    describe('accepts', () => {
+      it('with mayRead', async () => {
+        jest
+          .spyOn(service, 'mayRead')
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore
+          .mockImplementationOnce(async () => {
+            return true;
+          });
+        expect(
+          await service.checkPermissionOnNote(Permission.READ, user1, notes[0]),
+        ).toBeTruthy();
+      });
+      it('with mayWrite', async () => {
+        jest
+          .spyOn(service, 'mayWrite')
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore
+          .mockImplementationOnce(async () => {
+            return true;
+          });
+        expect(
+          await service.checkPermissionOnNote(
+            Permission.WRITE,
+            user1,
+            notes[0],
+          ),
+        ).toBeTruthy();
+      });
+      it('with isOwner', async () => {
+        jest
+          .spyOn(service, 'isOwner')
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore
+          .mockImplementationOnce(async () => {
+            return true;
+          });
+        expect(
+          await service.checkPermissionOnNote(
+            Permission.OWNER,
+            user1,
+            notes[0],
+          ),
+        ).toBeTruthy();
+      });
+    });
+    describe('denies', () => {
+      it('with no permission', async () => {
+        jest
+          .spyOn(service, 'mayRead')
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore
+          .mockImplementationOnce(async () => {
+            return false;
+          });
+        jest
+          .spyOn(service, 'mayWrite')
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore
+          .mockImplementationOnce(async () => {
+            return false;
+          });
+        jest
+          .spyOn(service, 'isOwner')
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore
+          .mockImplementationOnce(async () => {
+            return false;
+          });
```

Comment:
Why do you mock all the implementations, when you then only test read afterwards?

Reply:
changed to isOwner, because it's the last check.